python/cudf/cudf/tests/test_search.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+import pytest
+
+import cudf
+from cudf.tests.utils import assert_eq, gen_rand, random_bitmask
+
+
+@pytest.mark.parametrize("side", ["left", "right"])
+@pytest.mark.parametrize("obj_class", ["series", "index"])
+def test_searchsorted(side, obj_class):
+    nelem = 1000
+    column_data = gen_rand("float64", nelem)
+    column_mask = random_bitmask(nelem)
+
+    values_data = gen_rand("float64", nelem)
```

Comment:
It would be nice to support/test the case where `values_data` contains more elements than `column_data`.  I recently cloned this branch, and I am getting an `out-of-bound` error for the case we need in Dask (see snippet below).

In Dask, `searchsorted` is currently used (for pandas dataframes) to determine the partition of each element in a column, given a list of `divisions`.   For a cudf dataframe, the code might look something like this:

```python
values = cudf.Series([0, 4, 0, 4, 0, 4])  # Input: Series/column to calculate partition of each row
divisions = [0, 2, 4] # Input: Desired divisions

partitions = cudf.Series(divisions).searchsorted(values, side="right") - 1
```

It is often the case that the `values` argument of `searchsorted` will contain more elements than the base object (because the base object corresponds to the desired divisions, and `values` is an entire column).

Reply:
cc @mrocklin

Reply:
Can you give me a reproducer? Is the snippet you shared also causing the error?

Reply:
Sorry for the delay @devavret -  I think I was just a bit thrown off by some differences in behavior between cudf and pandas.  This python-code snippet illustrates these differences:

```python
import cudf, pandas as pd

# CuDF
values = cudf.Series([0, 4, 0, 4, 0, 4])
divisions = [0, 2, 4]
partitions = cudf.Series(divisions).searchsorted(values, side="right") - 1
print(type(partitions))

# Pandas
values_pd = values.to_pandas()
partitions_pd = pd.Series(divisions).searchsorted(values_pd, side="right") - 1
print(type(partitions_pd))

# Show CuDF results in len(index) != len(partitions._column)
print(len(partitions.index) == len(partitions._column))

# Printing result raises error (probably because of index issue):
print(partitions)
```

<details open>
<summary>Output:</summary>
<br>

```
<class 'cudf.dataframe.series.Series'>
<class 'numpy.ndarray'>
False

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-39-3a48a307b922> in <module>
     16 
     17 # Printing result raises error (probably because of index issue):
---> 18 print(partitions)

~/workspace/cudf-dask-devel/cudf-sort/python/cudf/cudf/dataframe/series.py in __str__(self)
    513 
    514     def __str__(self):
--> 515         return self.to_string(nrows=10)
    516 
    517     def __repr__(self):

~/workspace/cudf-dask-devel/cudf-sort/python/cudf/cudf/dataframe/series.py in to_string(self, nrows)
    504             dtypes=dtypes,
    505             more_rows=more_rows,
--> 506             series_spacing=True,
    507         )
    508         return (

~/workspace/cudf-dask-devel/cudf-sort/python/cudf/cudf/formatting.py in format(index, cols, dtypes, show_headers, more_cols, more_rows, min_width, series_spacing)
     96         out.append(cell_template.format(str(index.name), 0))
     97     for i in range(nrows):
---> 98         row = [cell_template.format(str(index[i]), 0)]
     99         for k, vs in cols.items():
    100             if k == lastcol:

~/workspace/cudf-dask-devel/cudf-sort/python/cudf/cudf/dataframe/index.py in __getitem__(self, index)
    393 
    394         elif isinstance(index, int):
--> 395             index = utils.normalize_index(index, len(self))
    396             index += self._start
    397             return index

~/workspace/cudf-dask-devel/cudf-sort/python/cudf/cudf/utils/utils.py in normalize_index(index, size, doraise)
     98         index = size + index
     99     if doraise and not (0 <= index < size):
--> 100         raise IndexError("out-of-bound")
    101     return min(index, size)
    102 

IndexError: out-of-bound
```

</details>

Note that my previous comment was just pointing out the error I get when trying to print `partitions`.

Reply:
Fixed in https://github.com/rapidsai/cudf/pull/2156/commits/82515adc5cabcd72f4c1c36538e66ccb65a7ed18. The fix was quick but for some reason all my conda environments broke today.

Reply:
Thanks @devavret !